flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,593 @@
+from __future__ import unicode_literals, division, absolute_import
+
+import json
+import logging
+from builtins import *
+from collections import MutableSet
+
+import requests
+from future.moves.urllib.parse import urlparse, quote
+from future.utils import python_2_unicode_compatible
+from requests import RequestException
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.qualities import Requirements
+
+log = logging.getLogger('radarr')
+
+
+@python_2_unicode_compatible
+class RadarrRequestError(Exception):
+    def __init__(self, value, logger=log, **kwargs):
+        super(RadarrRequestError, self).__init__()
+        # Value is expected to be a string
+        if not isinstance(value, str):
+            value = str(value)
+        self.value = value
+        self.log = logger
+        self.kwargs = kwargs
+
+    def __str__(self):
+        return self.value
+
+
+class RadarrMovieAlreadyExistsError(Exception):
+    def __init__(self):
+        super(RadarrMovieAlreadyExistsError, self).__init__()
+
+
+def spec_exception_from_response_ex(radarr_request_ex):
+    error_message = None
+
+    if "error_message" in radarr_request_ex.kwargs:
+        error_message = radarr_request_ex.kwargs["error_message"]
+
+    if not error_message:
+        return None
+
+    if error_message.lower() == "this movie has already been added":
+        return RadarrMovieAlreadyExistsError()
+
+    return None
+
+
+def request_get_json(url, headers):
+    """ Makes a GET request and returns the JSON response """
+    try:
+        response = requests.get(url, headers=headers)
```

Comment:
Shouldn't we always use `task.requests`?

Reply:
Yeah, is it useful for application APIs though? Would need a bit refactoring ..